File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def get_height(self, node: Optional[TreeNode]) -> int:
        if not node:
            return -1
        
        leftHeight = self.get_height(node.left)
        rightHeight = self.get_height(node.right)

        return 1 + max(leftHeight, rightHeight)

    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True

        leftHeight = self.get_height(root.left)
        rightHeight = self.get_height(root.right)

        res = abs(leftHeight - rightHeight)

        if res > 1:
            return False
        
        return self.isBalanced(root.left) and self.isBalanced(root.right)
            

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int len = numbers.length;
        int left = 0;
        int[] res = new int[2];
        while (left < len) {
            int right = left + 1;
            while (right < len) {
                if (numbers[left] + numbers[right] == target) {
                    res[0] = left + 1;
                    res[1] = right + 1;
                }
                right++;
            }
            left++;
        }
        return res;
    }
}
